frame/contracts/src/benchmarking/mod.rs

```diff
@@ -277,7 +277,7 @@ benchmarks! {
 		assert_eq!(StorageVersion::get::<Pallet<T>>(), 2);
 	}
 
-	// This benchmarks the weight of executing Migration::migrate when there are no migration in progress.
+	// This benchmarks the weight of executing 1 `NoopMigraton`
```

Comment:
no. this comment is valid for the benchmark `migration_noop`, but not here.
what you benchmark here is the `migrate` dispatchable, see my  [comment](https://github.com/paritytech/substrate/pull/14045#discussion_r1212116161)

Reply:
sorry might have overlooked, checking

Reply:
it's a bit confusing cause we use it for both the dispatchable and the migrate, but this is intended to bench this (https://github.com/paritytech/substrate/blob/d4349fb7a28142fd3dbdac92deef23b126f72c4a/frame/contracts/src/migration.rs#L350-L352)

When we get into this function we want to make sure that there is enough weight to at least get to that line so we can make useful progress.
https://github.com/paritytech/substrate/blob/d4349fb7a28142fd3dbdac92deef23b126f72c4a/frame/contracts/src/migration.rs#L370-L374

This might be slightly less than dispatching migrate, which adds some overhead related to dispatching a call, so we could add one more bench for that, but It does not really matter here, cause if you add too little you will essentially pay for a noop since T::migrate will exit early

Reply:
actually, I take that back, we should add one more benchmark for the dispatchable as even if not an attack vector, it would make the block weight slightly inaccurate.